libeventheader-decode-cpp/src/EventFormatter.cpp

```diff
@@ -345,16 +537,48 @@ class StringBuilder
         WriteBegin(DestWriteMax);
 
 #if _WIN32
-        auto const p = static_cast<uint16_t const*>(val);
-        WritePrintf(DestWriteMax,
-            "%x:%x:%x:%x:%x:%x:%x:%x",
-            bswap_16(p[0]), bswap_16(p[1]), bswap_16(p[2]), bswap_16(p[3]),
-            bswap_16(p[4]), bswap_16(p[5]), bswap_16(p[6]), bswap_16(p[7]));
+
+        using PfnRtlIpv6AddressToStringA = decltype(&RtlIpv6AddressToStringA);
+        static const auto pfnNotFound = reinterpret_cast<PfnRtlIpv6AddressToStringA>(static_cast<INT_PTR>(-1));
+        static PfnRtlIpv6AddressToStringA pfnStatic = nullptr;
+
+        auto pfn = pfnStatic;
+        if (pfn == nullptr)
+        {
+            auto const ntdll = GetModuleHandleW(L"ntdll.dll");
+            pfn = ntdll
+                ? reinterpret_cast<PfnRtlIpv6AddressToStringA>(GetProcAddress(ntdll, "RtlIpv6AddressToStringA"))
+                : nullptr;
+            if (pfn == nullptr)
+            {
+                pfn = pfnNotFound;
+            }
+
+            pfnStatic = pfn;
+        }
+
+        if (pfn == pfnNotFound)
```

Comment:
Note: For Windows, try to use `RtlIpv6AddressToStringA` if available to get properly canonicalized output. If not available, fall back to previous behavior of non-canonical but information-preserving output.